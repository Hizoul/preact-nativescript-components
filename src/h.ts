import * as preact from "preact"
import { Preact } from "preact-to-nativescript"

const customH: any = Preact.h

export default customH
