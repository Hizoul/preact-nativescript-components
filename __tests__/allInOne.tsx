import * as renderToJson from "preact-render-to-json"
import {
  AbsoluteLayout, ActionBar, ActionItem, ActivityIndicator, Button, DatePicker,
  DockLayout, FlexboxLayout, GridLayout,
  HtmlView,
  Image,
  Label,
  ListPicker,
  ListView,
  NavigationButton,
  Page,
  Placeholder,
  Progress,
  ScrollView,
  SearchBar,
  SegmentedBar,
  SegmentedBarItem,
  Slider, StackLayout,
  Switch,
  TabView,
  TabViewItem, TextField,
  TextView,
  TimePicker,
  WebView,
  WrapLayout
} from "../src"
import h from "../src/h"

test("All Components rendered in one", () => {
  expect(
    renderToJson(
      <Page>
        <ActionBar>
          <StackLayout>
            <Label>Custom Header</Label>
          </StackLayout>
          <ActionItem text="act" />
        <NavigationButton text="mybut" />
        </ActionBar>
        <StackLayout>
          <ActivityIndicator />
          <AbsoluteLayout />
          <DockLayout />
          <FlexboxLayout />
          <GridLayout />
          <Button />
          <DatePicker />
          <HtmlView />
          <Image />
          <Label />
          <ListPicker />
          <ListView />
          <Page />
          <Placeholder />
          <Progress />
          <ScrollView />
          <SearchBar />
          <SegmentedBar />
          <SegmentedBarItem />
          <Slider />
          <Switch />
          <TabView />
          <TabViewItem />
          <TextField />
          <TextView />
          <TimePicker />
          <WebView />
          <WrapLayout />
        </StackLayout>
      </Page>
    )
  ).toMatchSnapshot("all componts rendered in one")
})
