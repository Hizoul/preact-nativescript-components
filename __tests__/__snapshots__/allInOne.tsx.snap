// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Components rendered in one: all componts rendered in one 1`] = `
<page>
  <actionBar>
    <stackLayout>
      <label>
        Custom Header
      </label>
    </stackLayout>
    <actionItem
      text="act"
    />
    <navigationButton
      text="mybut"
    />
  </actionBar>
  <stackLayout>
    <activityIndicator />
    <absoluteLayout />
    <dockLayout />
    <flexboxLayout />
    <gridLayout />
    <button />
    <datePicker />
    <htmlView />
    <image />
    <label />
    <listPicker />
    <listView />
    <page />
    <placeholder />
    <progress />
    <scrollView />
    <searchBar />
    <segmentedBar />
    <segmentedBarItem />
    <slider />
    <switch />
    <tabView />
    <tabViewItem />
    <textField />
    <textView />
    <timePicker />
    <webView />
    <wrapLayout />
  </stackLayout>
</page>
`;
